src/validators/with_default.rs

```diff
@@ -81,13 +82,31 @@ impl BuildValidator for WithDefaultValidator {
 
         let sub_schema: &PyAny = schema.get_as_req(intern!(schema.py(), "schema"))?;
         let validator = Box::new(build_validator(sub_schema, config, definitions)?);
+
+        let copy_default = {
+            if let DefaultType::Default(default_obj) = &default {
+                let hash_value = default_obj
+                    .getattr(py, intern!(py, "__hash__"))
+                    .unwrap_or(py.NotImplemented());
+                hash_value.is_none(py) || hash_value.is(&py.NotImplemented())
```

Comment:
```suggestion
                default_obj.call_method0(intern!(py, "__hash__")).is_ok()
```

Is cleaner. 

If you just want to check that `default_obj` has the `__hash__` attribute, just use 

```rs
default_obj.hasattr(intern!(py, "__hash__"))?
```